src/components/editor/Bindings/Backfill/SectionWrapper.tsx

```diff
@@ -1,23 +1,48 @@
 import { Box, Stack, Typography } from '@mui/material';
+import TrialOnlyPrefixAlert from 'components/materialization/TrialOnlyPrefixAlert';
+import { useEntityType } from 'context/EntityContext';
 import { useIntl } from 'react-intl';
-import { BaseComponentProps } from 'types';
+import { useBinding_backfilledBindings } from 'stores/Binding/hooks';
+import { useBindingStore } from 'stores/Binding/Store';
+import { SectionWrapperProps } from './types';
 
-export default function SectionWrapper({ children }: BaseComponentProps) {
+export default function SectionWrapper({
+    alertMessageId,
+    bindingUUID,
+    children,
+}: SectionWrapperProps) {
     const intl = useIntl();
 
+    const entityType = useEntityType();
+
+    const backfillAllBindings = useBindingStore(
+        (state) => state.backfillAllBindings
+    );
+    const backfilledBindings = useBinding_backfilledBindings();
+
     return (
         <Box sx={{ mb: 4, mt: 3 }}>
-            <Typography
-                component="div"
-                style={{ marginBottom: 8 }}
-                variant="formSectionHeader"
-            >
-                {intl.formatMessage({
-                    id: 'workflows.collectionSelector.manualBackfill.header',
-                })}
-            </Typography>
+            <Stack spacing={1}>
+                <Typography component="div" variant="formSectionHeader">
+                    {intl.formatMessage({
+                        id: 'workflows.collectionSelector.manualBackfill.header',
+                    })}
+                </Typography>
+
+                {entityType === 'materialization' ? (
+                    <TrialOnlyPrefixAlert
+                        bindingUUID={bindingUUID}
+                        messageId={alertMessageId}
+                        triggered={
+                            bindingUUID
+                                ? backfilledBindings.includes(bindingUUID)
+                                : backfillAllBindings
+                        }
```

Comment:
To display an alert in the top-level _Backfill_ section when at least one aged, trial-only binding is backfilled, a hook must be defined which filters the set of resource configs to extract the UUIDs of trial-only bindings and the trigger condition will require comparing the two arrays.